### ISSUE TITLE
Finders

### ISSUE THREAD

**Issue by silentlight** [revolver-app/vuex-orm#14]
Hi @kiaking 

This is not an issue but something I wanted to share with you.
I was wondering why do we have those two `normal` selectors:
`const users = store.getters['entities/users/all']()` and
`const users = store.getters['entities/users/find'](1)` aside from query builder.

For me for instance, it would be much easier to have syntax like below for those two:
`const users = store.getters['entities/users/query']().all()` and
`const users = store.getters['entities/users/query']().find(1)`.

This way everything inside parents stays always the same for all selectors and only the last part changes based on the type of the operation you want to perform. 
What are your thoughts on this one? Does it make sense?

**Comment by silentlight** [comment]
Ok I think I found answer to my own question. For getting all records I can just use:
`const users = store.getters['entities/users/query']().get()` and for finding single record by primary_key I can just do `const users = store.getters['entities/users/query']().first(1)`.
Coming from Ruby world I thought that the number we provide to `first` means how many first items do we want to return :)

**Comment by kiaking** [comment]
Hi @silentlight. Thanks for your feedback! I appreciate it. It makes Vuex ORM become a much better library.

You have a good point, and I think I'll reconsider the API.

> I was wondering why do we have those two normal selectors:

The reason I made those two selectors – `all` and `find` – is because I was trying to adapt to [Laravel](https://laravel.com) (Framework like rails for PHP) API. Laravel has exactly same `all` and `find` method ([doc can be found here](https://laravel.com/docs/5.5/eloquent#retrieving-models)). Same for `get` and `first`, and how query builder can chain methods. In fact, Vuex ORM has very similar API to Laravel's ORM (because I am a big Laravel user).

However, I think it's true that your proposal seems easier to understand and more consistent. Honestly, I wasn't thinking that much when I first wrote that two methods :p

To take it even farther, if everything is chained by `query`, maybe we don't need `query`. Just directly call submodule like this.

```js
// No `query`, just call submodule. Not sure if Vuex can do this
// but I think it's worth investigating.
store.getters['entities/users']().get()
```

I'm not sure if I'll want to remove "parent" method such as  `getters['entities/users/all']`, but I'll definitely make `all` and `find` methods callable after `query` as you proposed.

> Coming from Ruby world I thought that the number we provide to first means how many first items do we want to return :)

Thx for the info! I didn't know that, but it did remind me to add such `limit` feature for the query builder.

Initially, `first` method wasn't intended to take id as an argument. Because you have `find` method for that purpose. `first` method was more intended to use in conjunction with `where` clauses such as;

```js
store.getters['entities/users/query']().where('age', 24).first()
```

So, next action, I'll consider a bit more, and will make PR for adding new API then mention you in that PR :)
